backend/src/litigation_api/ingestion/pipeline.py

```
@@ -0,0 +1,121 @@
+logger = logging.getLogger(__name__)
+
+SUPPORTED_DOCUMENT_TYPES = {"Complaint", "Opinion/Order", "Settlement"}
+
+class IngestionPipeline:
+    def __init__(self, s3_endpoint_url: str, s3_bucket_name: str, client: Optional[CRLCAClient] = None):
+        self.s3_endpoint_url = s3_endpoint_url
+        self.s3_bucket_name = s3_bucket_name
+        self.client = client
+        self.s3_client = boto3.client("s3", endpoint_url=self.s3_endpoint_url)
+
+    def _ensure_bucket_sync(self):
```

Ruff (0.15.11)

[warning] Missing return type annotation for private function `_ensure_bucket_sync`

Add return type annotation: `None`

(ANN202)